web/app/fr/gospeak/web/pages/speaker/SpeakerCtrl.scala

```diff
@@ -1,26 +1,52 @@
 package fr.gospeak.web.pages.speaker
 
+import java.time.Instant
+
+import cats.effect.IO
 import com.mohiva.play.silhouette.api.Silhouette
+import com.mohiva.play.silhouette.api.actions.SecuredRequest
 import fr.gospeak.core.domain.User
-import fr.gospeak.core.services.storage.{UserGroupRepo, UserTalkRepo}
+import fr.gospeak.core.services.storage.{UserGroupRepo, UserRepo, UserTalkRepo}
 import fr.gospeak.web.auth.domain.CookieEnv
 import fr.gospeak.web.domain.Breadcrumb
-import fr.gospeak.web.pages.speaker.SpeakerCtrl._
 import fr.gospeak.web.pages.user.UserCtrl
 import fr.gospeak.web.utils.UICtrl
-import play.api.mvc.{Action, AnyContent, ControllerComponents}
+import play.api.data.Form
+import play.api.mvc.{Action, AnyContent, ControllerComponents, Result}
 
 class SpeakerCtrl(cc: ControllerComponents,
                   silhouette: Silhouette[CookieEnv],
                   groupRepo: UserGroupRepo,
-                  talkRepo: UserTalkRepo) extends UICtrl(cc, silhouette) {
+                  talkRepo: UserTalkRepo,
+                  userRepo: UserRepo) extends UICtrl(cc, silhouette) {
 
   import silhouette._
 
-  def profile(): Action[AnyContent] = SecuredAction { implicit req =>
-    val b = breadcrumb(req.identity.user)
-    Ok(html.profile()(b))
+
+  def getProfile(): Action[AnyContent] = SecuredAction.async { implicit req =>
+    val b = SpeakerCtrl.breadcrumb(req.identity.user)
+    val form = ProfileForms.create
+    val filledForm = if (form.hasErrors) {
+      println(s"form.has Error : ${form.errors}")
+      form
+    } else form.fill(req.identity.user.editable)
+    IO(Ok(html.profile(filledForm)(b))).unsafeToFuture()
+  }
+
+  def profile(): Action[AnyContent] = SecuredAction.async { implicit req =>
+    val now = Instant.now()
+    ProfileForms.create.bindFromRequest.fold(
+      formWithErrors => doEditOrCreate(formWithErrors),
+      data => userRepo.edit(req.identity.user, data, now).map { _ => Redirect(routes.SpeakerCtrl.profile()) }
```

Comment:
tu peux ajouter un message de succès
```suggestion
      data => userRepo.edit(req.identity.user, data, now).map { _ => Redirect(routes.SpeakerCtrl.profile()).flashing("success" -> "Profile updated") }
```